src/main/java/com/github/manolo8/darkbot/core/manager/HangarManager.java

```diff
@@ -0,0 +1,222 @@
+package com.github.manolo8.darkbot.core.manager;
+
+import com.github.manolo8.darkbot.Main;
+import com.github.manolo8.darkbot.core.entities.Dron;
+import com.github.manolo8.darkbot.core.entities.Hangar;
+import com.github.manolo8.darkbot.core.objects.swf.Array;
+import com.google.gson.JsonArray;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParser;
+
+import java.io.BufferedReader;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.net.HttpURLConnection;
+import java.net.URL;
+import java.util.ArrayList;
+import java.util.Base64;
+
+import static com.github.manolo8.darkbot.Main.API;
+
+public class HangarManager {
+
+    private final Main main;
+    private boolean disconnecting = false;
+    private Character exitKey = 'l';
+    private long lastChangeHangar = 0;
+    private long disconectTime = 0;
+    private ArrayList<Hangar> hangars;
+    private ArrayList<Dron> drones;
+
+    /*
+     * Para sacar los vants
+     * /flashAPI/inventory.php/action=getHangar&params=
+     * Params es:
+     * {"params":{"hi": HANGARID}}
+     * En base 64
+     */
+
+    public HangarManager(Main main){
+        this.main = main;
+        this.hangars = new ArrayList<Hangar>();
+        this.drones = new ArrayList<Dron>();
+    }
+
+    public boolean changeHangar(String hangarID) {
+        HttpURLConnection conn = null;
+        if (!this.disconnecting) {
+            disconnect();
+        }
+        if (this.lastChangeHangar <= System.currentTimeMillis() - 40000 && this.main.backpage.sidStatus().contains("OK") || !hangarID.isEmpty()) {
+            if (this.disconectTime <= System.currentTimeMillis() - 20000) {
+                String instance = this.main.statsManager.instance, sid = this.main.statsManager.sid;
+                if (instance == null || instance.isEmpty() || sid == null || sid.isEmpty()) return false;
+                String url = instance + "/indexInternal.es?action=internalDock&subAction=changeHangar&hangarId=" + hangarID;
+                try {
+                    conn = (HttpURLConnection) new URL(url)
```

Comment:
Use a common method in BackpageManager to create requests, instead of copy-pasting http-request-creation code arround

Reply:
Changed